src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringTests.cs

```diff
@@ -0,0 +1,405 @@
+        public async Task TestNonUnitString()
+        {
+            const string nonUnitString_cs = @" 
+using System; 
+using System.Text;
+ 
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            StringBuilder sb = new StringBuilder();
+            const string ch = ""ab"";
+            sb.Append(ch);
+        } 
+    } 
+}";
+            await VerifyCS.VerifyAnalyzerAsync(nonUnitString_cs);
+            const string nonUnitString_vb = @"
+Imports System
+
+Module Program
+    Class TestClass
+        Public Sub Main(args As String())
+            Const ch As String = ""ab""
+            Dim builder As New System.Text.StringBuilder
+            builder.Append(ch)
+        End Sub
+    End Class
+End Module
+";
+            await VerifyVB.VerifyAnalyzerAsync(nonUnitString_vb);
+        }
+
+        [Fact]
+        public async Task TestNoCallToStringAppend()
+        {
+            const string noCallToStringAppend_cs = @" 
+using System; 
+using System.Text;
+ 
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            StringBuilder sb = new StringBuilder();
+            const string ch = ""a"";
+        } 
+    } 
+}";
+            await VerifyCS.VerifyAnalyzerAsync(noCallToStringAppend_cs);
+
+            const string noCallToStringAppend_vb = @"
+Imports System
+
+Module Program
+    Class TestClass
+        Public Sub Main(args As String())
+            Const ch As String = ""a""
+            Dim builder As New System.Text.StringBuilder
+        End Sub
+    End Class
+End Module
+";
+            await VerifyVB.VerifyAnalyzerAsync(noCallToStringAppend_vb);
+        }
+
+        [Fact]
+        public async Task TestNonConstUnitString()
+        {
+            const string nonConstUnitString_cs = @" 
+using System; 
+using System.Text;
+ 
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            StringBuilder sb = new StringBuilder();
+            string ch = ""ab"";
+            sb.Append(ch);
+        } 
+    } 
+}";
+            await VerifyCS.VerifyAnalyzerAsync(nonConstUnitString_cs);
+
+            const string nonConstUnitString_vb = @"
+Imports System
+
+Module Program
+    Class TestClass
+        Public Sub Main(args As String())
+            Dim ch As String = ""a""
+            Dim builder As New System.Text.StringBuilder
+            builder.Append(ch)
+        End Sub
+    End Class
+End Module
+";
+            await VerifyVB.VerifyAnalyzerAsync(nonConstUnitString_vb);
+        }
+
+        [Fact]
+        public async Task TestAppendLiteral()
+        {
+            const string appendLiteralInput_cs = @" 
+using System; 
+using System.Text;
+ 
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            StringBuilder sb = new StringBuilder();
+            sb.Append([|"",""|]);
```

Comment:
Can you add no diagnostic tests for:
1. Multi character string: `sb.Append(""ab"");`
2. Null literal: `sb.Append(null);`
3. Interpolated string: `sb.Append($"{someLocal}");`

Reply:
Done.